sdk/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -136,6 +159,177 @@ void LogExcessive(string message)
             }
         }
 
+        private static async Task<Pulumirpc.Callback> AllocateTransform(Callbacks callbacks, ResourceTransform transform)
+        {
+            var wrapper = new Callback(async (message, token) =>
+            {
+                var request = Pulumirpc.TransformRequest.Parser.ParseFrom(message);
+
+                var props = ImmutableDictionary.CreateBuilder<string, object?>();
+                foreach (var kv in request.Properties.Fields)
+                {
+                    var outputData = Serialization.Deserializer.Deserialize(kv.Value);
+                    // If it's plain just use the value directly
```

Comment:
What if there are dependencies? I think we need to keep it an output in that case.